src/main/kotlin/com/mineinabyss/blocky/BlockyCommandExecutor.kt

```diff
@@ -78,9 +76,10 @@ class BlockyCommandExecutor : IdofrontCommandExecutor(), TabCompleter {
                     }
 
                     item.editItemMeta {
-                        (this as? LeatherArmorMeta)?.setColor(color.toColor)
-                            ?: (this as? PotionMeta)?.setColor(color.toColor)
-                            ?: (this as? MapMeta)?.setColor(color.toColor) ?: return@playerAction
+                        (asRGBColorable() ?: run {
+                            player.error("This item cannot be dyed.")
+                            return@playerAction
```

Comment:
u can just return here without a run
```kt
asColorable()?.setColor(color.toColor) ?: return@playerAction player.error("This item cannot be dyed.")
```
 or something like that
should also move this into Idofront and make it also extend itemstack
ItemStack.dye or osmething idk